File: tutorials/python/tutorial_deposition.py
from semiconductor_simulator import Simulator

print("Tutorial: Deposition Module")
simulator = Simulator()
print("Step 1: Initialize grid")
simulator.initialize_geometry(10, 10)
print("Step 2: Apply photoresist pattern")
mask = [[1, 0, 1], [0, 1, 0], [1, 0, 1]]
simulator.run_exposure(13.5, 0.33, mask)
print("Step 3: Run uniform deposition")
simulator.run_deposition(0.1, "SiO2", "uniform")
print("Step 4: Visualize result")
simulator.initialize_renderer(400, 400)
simulator.render(simulator.get_wafer())

File: tutorials/cpp/tutorial_deposition.cpp
#include "../../src/cpp/core/wafer.hpp"
#include "../../src/cpp/modules/geometry/geometry_manager.hpp"
#include "../../src/cpp/modules/photolithography/lithography_model.hpp"
#include "../../src/cpp/modules/deposition/deposition_model.hpp"
#include <memory>
#include <iostream>

int main() {
    std::cout << "Tutorial: Deposition Module\n";
    auto wafer = std::make_shared<Wafer>(300.0, 775.0, "silicon");
    GeometryManager geometry_manager;
    std::cout << "Step 1: Initialize grid\n";
    geometry_manager.initializeGrid(wafer, 10, 10);
    LithographyModel lithography;
    std::vector<std::vector<int>> mask = {{1, 0, 1}, {0, 1, 0}, {1, 0, 1}};
    std::cout << "Step 2: Apply photoresist pattern\n";
    lithography.simulateExposure(wafer, 13.5, 0.33, mask);
    DepositionModel deposition;
    std::cout << "Step 3: Run uniform deposition\n";
    deposition.simulateDeposition(wafer, 0.1, "SiO2", "uniform");
    std::cout << "Film layers: " << wafer->getFilmLayers().size() << "\n";
    return 0;
}